src/Microsoft.OData.Core/ODataResource.cs

```diff
@@ -183,7 +184,11 @@ public IEnumerable<ODataFunction> Functions
         public IEnumerable<ODataProperty> Properties
         {
             get { return this.MetadataBuilder.GetProperties(this.properties); }
-            set { this.properties = value; }
+            set
+            {
+                VerifyProperties(value);
+                this.properties = value;
+            }

```

Comment:
This will prevent me from passing in a list of properties that contains an ODataResourceValue (or collection containing an ODataResourceValue), but it won't prevent me from passing in a List<ODataProperty> that I later add an ODataResourceValue to, correct? Should we also validate on get, to make sure an ODataResourceValue hasn't been added to the list or a collection property after it was set?

Reply:
@mikepizzo I updated this PR with a new commit. Would you please take a look?

Reply:
Still seems to have same issue; it appears that I can do:

  List<Properties> properties = new List<Properties>();
  myODataResource.Properties = properties; //passes validation as there are not (yet) any ODataResourceValue properties
  properties.Add(new ODataProperty("foo",new ODataResourceValue(....));   //now I've added an ODataResourceValue property; does this get flagged anywhere?

Will we add checks, i.e. when writing a top-level resource, that again validates that the Properties collection does not contain an ODataResourceValues?


---
In reply to: [236774022](https://github.com/OData/odata.net/pull/1331#discussion_r236774022) [](ancestors = 236774022)